api/src/modules/projects/projects.service.ts

```diff
@@ -21,14 +29,70 @@ export class ProjectsService extends AppBaseService<
 
   async extendFindAllQuery(
     query: SelectQueryBuilder<Project>,
-    fetchSpecification: FetchSpecification,
+    fetchSpecification: ExtendedFetchSpecification,
   ): Promise<SelectQueryBuilder<Project>> {
     // Filter by project name
     if (fetchSpecification?.filter?.projectName) {
       query = query.andWhere('project_name ILIKE :projectName', {
         projectName: `%${fetchSpecification.filter.projectName}%`,
       });
     }
+
+    // Filter by abatement potential
+    if (this.isAbatementPotentialFilterValid(fetchSpecification)) {
```

Comment:
Ideally, we should perform all validations in the controller using zod. We have some issues with this specific entity, maybe let me show you the issue and you might know what's going on. If you don't know the issue straight away, let's keep these validations here